tests/periph_pm/main.c

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (C) 2016-2018 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Power management peripheral test.
+ *
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "periph/pm.h"
+#ifdef MODULE_PM_LAYERED
+#ifdef MODULE_PERIPH_RTC
+#include "periph/rtc.h"
+#endif
+#include "pm_layered.h"
+#endif
+#include "shell.h"
+
+#ifdef MODULE_PM_LAYERED
+static int parse_mode(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("Error: power mode expected (0 - %d).\n", PM_NUM_MODES - 1);
+        return -1;
+    }
+
+    int mode = atoi(argv[1]);
+
+    if (mode < 0 || mode >= (int) PM_NUM_MODES) {
+        printf("Error: power mode not in range 0 - %d.\n", PM_NUM_MODES - 1);
+        return -1;
+    }
+
+    return mode;
+}
+
+static int parse_duration(int argc, char **argv)
+{
+    if (argc < 3) {
+        printf("Error: duration expected (0 - %d).\n", INT_MAX);
+        return -1;
+    }
+
+    int duration = atoi(argv[2]);
+
+    if (duration < 0) {
+        puts("Error: duration must be a positive number.");
+        return -1;
+    }
+
+    return duration;
+}
+
+#ifdef MODULE_PERIPH_RTC
+static void cb_rtc(void *arg)
+{
+    int level = (int)arg;
+
+    pm_block(level);
+}
+#endif /* MODULE_PERIPH_RTC */
+#endif /* MODULE_PM_LAYERED */
+
+static int cmd_off(int argc, char **argv)
```

Comment:
`cmd_off` and `cmd_reboot` should be move to the top and then all functions that requires MODULE_PM_LAYERED could be grouped in the same conditional `#ifdef`.

Reply:
I'd like to keep it this way: I now have the commands grouped together, and utility functions on top.